src/Microsoft.VisualStudio.Web.CodeGeneration.Design/Program.cs

```diff
@@ -0,0 +1,126 @@
+﻿using Microsoft.Extensions.CommandLineUtils;
+using Microsoft.VisualStudio.Web.CodeGeneration.Utils.Messaging;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Web.CodeGeneration.ProjectInfo;
+
+namespace Microsoft.VisualStudio.Web.CodeGeneration.Design
+{
+    public class Program
+    {
+        public const string TOOL_NAME = "dotnet-aspnet-codegenerator-design";
+
+        private static ConsoleLogger _logger;
+
+        private const string APPNAME = "Code Generation";
+        private const string APP_DESC = "Code generation for Asp.net Core";
+
+        public static void Main(string[] args)
+        {
+            _logger = new ConsoleLogger();
+            _logger.LogMessage($"Command Line: {string.Join(" ", args)}", LogMessageLevel.Trace);
+
+            //DotnetToolDispatcher.EnsureValidDispatchRecipient(ref args);
+            Execute(args, _logger);
+        }
+
+        private static void Execute(string[] args, ConsoleLogger _logger)
+        {
+            var app = new CommandLineApplication(false)
+            {
+                Name = APPNAME,
+                Description = APP_DESC
+            };
+
+            // Define app Options;
+            app.HelpOption("-h|--help");
+            var projectPath = app.Option("-p|--project", "Path to project.json", CommandOptionType.SingleValue);
+            var packagesPath = app.Option("-n|--nuget-package-dir", "Path to check for Nuget packages", CommandOptionType.SingleValue);
+            var appConfiguration = app.Option("-c|--configuration", "Configuration for the project (Possible values: Debug/ Release)", CommandOptionType.SingleValue);
+            var framework = app.Option("-tfm|--target-framework", "Target Framework to use. (Short folder name of the tfm. eg. net451)", CommandOptionType.SingleValue);
```

Comment:
Suggestion: might be nice to stick with the CLI convention for this: "-f|--framework"